src/main/java/org/cryptomator/cryptofs/CryptoFileSystemModule.java

```diff
@@ -37,6 +39,20 @@ public Cryptor provideCryptor(CryptorProvider cryptorProvider, @PathToVault Path
 		}
 	}
 
+	private String generateFileIdIfPossible(byte[] fileBytes) {
```

Comment:
Drop `IfPossible`. Why shouldn't it be possible? See comment below regarding handling of `NoSuchAlgorithmException`.